src/socket/udp_socket.cr

```diff
@@ -88,4 +88,139 @@ class UDPSocket < IPSocket
     bytes_read, sockaddr, addrlen = recvfrom(message)
     {bytes_read, IPAddress.from(sockaddr, addrlen)}
   end
+
+  # Reports whether transmitted multicast packets should be copied and sent
+  # back to the originator.
+  def multicast_loopback?
+    case @family
+    when Family::INET
+      optname = LibC::IP_MULTICAST_LOOP
```

Comment:
It would be better if you simply duplicate the call to `getsockopt_bool` in both branches using the constants directly as arguments. This makes the code shorter overall and easier to follow. Ditto all below.